moco-switchover-downtime-monitor/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+)
+
+const httpServerPort = 8080
+
+var flagNamespace = pflag.String("namespace", "", "The namespace this program runs in. default: read from serviceaccount file")
+
+var currentNamespace = "default"
+
+func newZapLogger() *zap.Logger {
+	logger, err := zap.NewProduction()
+	if err != nil {
+		panic(err)
+	}
+	return logger
+}
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	logger := newZapLogger()
+	defer logger.Sync()
+
+	pflag.Parse()
+	args := pflag.Args()
+
+	mysqlClusterNames := args
+
+	if len(mysqlClusterNames) == 0 {
+		logger.Error("no MySQLCluster is specified")
+		return 1
+	}
+
+	if *flagNamespace != "" {
+		currentNamespace = *flagNamespace
+	} else {
+		ns, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		if err != nil {
+			logger.Error("could not read namespace file", zap.Error(err))
+			return 1
+		}
+		currentNamespace = string(ns)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	wg := sync.WaitGroup{}
+
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
In this case, `signal.NotifyContext` can be used.
https://pkg.go.dev/os/signal#NotifyContext

Reply:
I wanted to log what signal is received.